src/lib/components/Assessment/index.js

```diff
@@ -0,0 +1,232 @@
+    this.onOpenClick = this.onOpenClick.bind(this);
+    this.openAssessment = this.openAssessment.bind(this);
+    this.closeAssessment = this.closeAssessment.bind(this);
+  }
+
+  render() {
+    if (!this.prerenderedChildren) {
+      this.prerenderedChildren = [];
+      this.setLeader = [];
+
+      for (const child of this.children) {
+        if (child.classList.contains("web-assessment__set-leader")) {
+          this.setLeader.push(child);
+        } else {
+          this.prerenderedChildren.push(child);
+        }
+      }
+    }
+
+    return html`
+      <div class="w-callout__header web-assessment__header">
+        <h2 class="w-callout__lockup web-assessment__lockup">
+          Check your understanding
+        </h2>
+        ${this.setLeader}
+        <button
+          @click="${this.onCloseClick}"
+          class="w-button--icon w-button--round web-assessment__close"
+          data-icon="close"
+        >
+          <span role="tooltip" class="w-tooltip">
+            Close
+          </span>
+        </button>
+      </div>
+      ${this.prerenderedChildren}
+    `;
+  }
+
+  firstUpdated() {
+    this.classList.remove("unresolved");
+    // Override BaseModalElement's inert behavior since Assessment
+    // is visible on desktop in closed state.
+    // (display: none used to remove it from the tab order when closed on mobile.)
+    this.inert = false;
+    // Render the launcher that appears in closed state on mobile.
+    this.renderLauncher();
+    // Listen to reset requests from child question components.
+    const questions = this.querySelectorAll("web-question");
+
+    for (const question of questions) {
+      question.addEventListener("request-assessment-reset", this.reset);
+    }
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    // Close modal if viewport is >481 px so document isn't inert on desktop
+    const mqString = `(min-width: 481px)`;
+
+    matchMedia(mqString).addListener(this.onAssessmentResize);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    const mqString = `(min-width: 481px)`;
+
+    matchMedia(mqString).removeListener(this.onAssessmentResize);
+  }
+
+  renderLauncher() {
+    const launcher = document.createElement("div");
+    const contentTemplate = (setLeader) => html`
+      <div class="w-callout__header web-assessment__header">
+        <h2 class="w-callout__lockup web-assessment__lockup">
+          Check your understanding
+        </h2>
+        <div class="w-callout__blurb web-assessment__set-leader">
+          ${setLeader}
+        </div>
+      </div>
+      <button
+        @click="${this.onOpenClick}"
+        class="w-button w-button--primary web-assessment__button web-assessment__open"
+      >
+        Open quiz
+      </button>
+    `;
+    // lit-element prevents children from being duplicated,
+    // so grab setLeader text content.
+    const text = this.setLeader[0] ? this.setLeader[0].innerText : "";
```

Comment:
nit: don't use `.innerText` (it has weird issues and is an IE-introduced property), always `.textContent`